### ISSUE TITLE
JSON deserialization fails for collapsed procedure call blocks containing at least one input variable

### ISSUE THREAD

**Issue by cmjcharlton** [google/blockly#6091]
**Describe the bug**

If I attempt to load a workspace from JSON containing a collapsed procedure call block with at least one input variable then this block is missing from the loaded workspace.

**To Reproduce**

1. Open the Blockly playground
2. Click 'Functions' in the toolbox
3. Drag a procedure block into the workspace
4. Open its mutator and add an input
5. Right-click the block and select 'Create "do something"'
6. Right-click the created procedure call block and collapse it
7. Click 'Save JSON'
8. Click 'Load'

**Expected behavior**

Both the procedure definition and call procedure blocks should be present in the loaded workspace.

**Screenshots**

Workspace to be saved:
![image](https://user-images.githubusercontent.com/90400333/164227624-4a820db9-067c-464d-ae58-abff22fb0905.png)

Workspace after load:
![image](https://user-images.githubusercontent.com/90400333/164227692-b95adb60-f7d4-4f9f-a36f-b60e4c575c8c.png)

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - Browser [Chrome, Firefox]
 - Version [Chrome 100.0.4896.127 (Official Build) (64-bit), Firefox 99.0.1 (64-bit)]

**Stack Traces**

```
block.js:2022 Uncaught Error: Input not found: _TEMP_COLLAPSED_INPUT
    at BlockSvg.module$exports$Blockly$Block.Block.removeInput (block.js:2022:11)
    at BlockSvg.module$exports$Blockly$BlockSvg.BlockSvg.removeInput (block_svg.js:1376:27)
    at BlockSvg.module$exports$Blockly$BlockSvg.BlockSvg.updateCollapsed_ (block_svg.js:673:12)
    at BlockSvg.module$exports$Blockly$BlockSvg.BlockSvg.setCollapsed (block_svg.js:647:12)
    at BlockSvg.setProcedureParameters_ (procedures.js:739:10)
    at BlockSvg.loadExtraState (procedures.js:899:12)
    at module$contents$Blockly$serialization$blocks_loadExtraState (blocks.js:452:11)
    at module$contents$Blockly$serialization$blocks_appendPrivate (blocks.js:394:3)
    at module$exports$Blockly$serialization$blocks.appendInternal (blocks.js:345:17)
    at module$exports$Blockly$serialization$blocks.append (blocks.js:308:10)
```

**Additional context**

Although it involves collapsed blocks I don't believe that this is the exactly the same as issue #6076 as it appears to be related to extraState rather than icon visibility, however again the error does not appear to occur if saving/loading via XML. It also does not occur for either deserialization method if the procedure call block contains no input variables.

**Comment by tweini** [comment]
Hi! I just finished the fix for #6076 and can confirm, that it doesn't fix this problem. I am into it and can provide a fix within a few days, stay tuned 😄

**Comment by cmjcharlton** [comment]
Thanks very much, that would be fantastic. I am currently working around both problems by pre-processing the JSON to expand collapsed blocks prior to deserialization, and then collapsing them again after the workspace is rendered. It will be nice to be able to remove my hack once this is fixed.

**Comment by tweini** [comment]
Hi @cmjcharlton bug is identified, fix follows. If you have some time, maybe you can do some manual test for my provided fix for #6076? Maybe you can find edge cases I haven't think of.

**Comment by gonfunko** [comment]
Thank you both for reporting and investigating this! I can repro the behavior you described.

**Comment by tweini** [comment]
Both types (i.e. "procedures_callnoreturn" and procedures_callreturn" ) are affected. The Precondition is: Call block contains **at least** one input variable

**Comment by cmjcharlton** [comment]
Thanks @tweini for pointing out my off-by-one error in the title/description (now corrected). I'm not sure how I managed to do that given my replication example only included one input.

**Comment by tweini** [comment]
fixed ✅  @BeksOmega, @cmjcharlton, who wants to close the issue?

**Comment by cmjcharlton** [comment]
I just tested the develop branch of Blockly with my application and can confirm that this appears to fix the problem that I was seeing. I would therefore be happy to see the issue closed. Thanks very much for finding the cause, and fixing it.

**Comment by BeksOmega** [comment]
Closed by #6103

**Comment by BeksOmega** [comment]
@cmjcharlton We've released patches for v7 and v8 containing this fix (and the fix to #6076) if you want to give them a try =)

**Comment by cmjcharlton** [comment]
Thanks @BeksOmega for the new update. I updated to the new patched version and removed the workarounds that I previously had in my application. I am no longer seeing the errors that I did before and everything seems to be working so far. I also checked the v7 backport version and that seems to be fixed for me too.
